gson/src/test/java/com/google/gson/GsonBuilderTest.java

```diff
@@ -254,4 +255,32 @@ public void testSetStrictness() throws IOException {
     assertThat(gson.newJsonReader(new StringReader("{}")).getStrictness()).isEqualTo(STRICTNESS);
     assertThat(gson.newJsonWriter(new StringWriter()).getStrictness()).isEqualTo(STRICTNESS);
   }
+
+  @Test
+  public void testRegisterTypeAdapterForObjectAndJsonElements() {
+    Type[] types = {
+            Object.class,
+            JsonElement.class,
+            JsonArray.class,
+    };
+    GsonBuilder gsonBuilder = new GsonBuilder();
+    for (Type type : types) {
+      assertThrows(IllegalArgumentException.class, () ->  gsonBuilder.registerTypeAdapter(type, NULL_TYPE_ADAPTER));
+    }
+  }
+
+
+  @Test
+  public void testRegisterTypeHierarchyAdapterJsonElements() {
+    Type[] types = {
+            JsonElement.class,
+            JsonArray.class,
+    };
+    GsonBuilder gsonBuilder = new GsonBuilder();
+    for (Type type : types) {
+      assertThrows(IllegalArgumentException.class,
+              () ->  gsonBuilder.registerTypeHierarchyAdapter((Class<?>) type, NULL_TYPE_ADAPTER));

```

Comment:
Looks like there is one space too much:
```suggestion
              () -> gsonBuilder.registerTypeHierarchyAdapter((Class<?>) type, NULL_TYPE_ADAPTER));

```